### PR TITLE
fix: Cleans up lighter code a tiny bit and fixes colored lighters becoming green

### PR DESCRIPTION
**Pull request by Spaghetti-bit**
## What Does This PR Do

Fixes https://github.com/ParadiseSS13/Paradise/issues/23111

## Why It's Good For The Game

Bugs bad.

## Images of changes

image
image

## Testing

Flicked a randomly colored lighter on and off, it stayed the same color.
## Changelog
:cl:
fix: Fixed lighters becoming green if they were not green before.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/lighters.dm [hunk 1]
```diff
@@ -19,10 +19,15 @@
 	/// Cooldown until the next turned off message/sound can be activated
 	var/next_off_message
 
-/obj/item/lighter/random/New()
+/obj/item/lighter/random
+	/// Random color that is assigned whenever we initialize a new random color lighter.
+	var/random_color = "g"
```
Comment (Contrabang) [inline code/game/objects/items/weapons/lighters.dm:24]: Im pretty sure we've removed this in a previous PR. Maybe you could try something like `base_icon_state` instead?

File: code/game/objects/items/weapons/lighters.dm [hunk 2]
```diff
@@ -19,10 +19,15 @@
...
+
+/obj/item/lighter/random/Initialize(mapload)
 	..()
```
Comment (DGamerL) [inline code/game/objects/items/weapons/lighters.dm:27]:
```suggestion
	. = ..()
```
This is something that `Initialize()` requires, and `New()` doesn't

File: code/game/objects/items/weapons/lighters.dm (hunk 1 above)
Comment (DGamerL) [reply code/game/objects/items/weapons/lighters.dm:24]: We didn't, the PR got closed
You can take a look at #23132 for a great example on how to fix this!

File: code/game/objects/items/weapons/lighters.dm [hunk 3]
```diff
@@ -18,11 +18,18 @@
 	var/next_on_message
 	/// Cooldown until the next turned off message/sound can be activated
 	var/next_off_message
+	/// Our base state for the lighter. This is for random color lighters
+	var/base_item_state = null
```
Comment (Contrabang) [inline code/game/objects/items/weapons/lighters.dm:22]: I dont see any cases where this is different from `base_icon_state`. Could we consolidate this just into `base_icon_state`?

Comment (Spaghetti-bit) [reply code/game/objects/items/weapons/lighters.dm:22]: Sure can, let me axe it.

File: code/game/objects/items/weapons/lighters.dm [hunk 4]
```diff
@@ -18,11 +18,15 @@
...
-/obj/item/lighter/random/New()
-	..()
-	var/color = pick("r","c","y","g")
-	icon_state = "lighter-[color]"
+/obj/item/lighter/random/Initialize(mapload)
+	. = ..()
+	lighter_color = pick("r","c","y","g")
+	base_icon_state = "lighter"
+	update_icon_state()
+	update_overlays()
```
Comment (Contrabang) [inline code/game/objects/items/weapons/lighters.dm:29]:
```suggestion
	update_icon()
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

**Review by Contrabang — APPROVED** [review]
looks cursed to me

File: code/game/objects/items/weapons/lighters.dm [hunk 5]
```diff
@@ -18,11 +18,14 @@
 	var/next_on_message
 	/// Cooldown until the next turned off message/sound can be activated
 	var/next_off_message
+	/// Our lighter color suffix. => [base_icon_state]-[lightercolor] => lighter-r
+	var/lighter_color = null
```
Comment (DGamerL) [inline code/game/objects/items/weapons/lighters.dm:22]:
```suggestion
	var/lighter_color
```

File: code/game/objects/items/weapons/lighters.dm [hunk 6]
```diff
@@ -18,11 +18,14 @@
...
 
-/obj/item/lighter/random/New()
-	..()
-	var/color = pick("r","c","y","g")
-	icon_state = "lighter-[color]"
+/obj/item/lighter/random/Initialize(mapload)
+	. = ..()
+	lighter_color = pick("r","c","y","g")
+	base_icon_state = "lighter"
```
Comment (DGamerL) [inline code/game/objects/items/weapons/lighters.dm:27]: Why is this in `Initialize()` and not on the variable on the lighter itself

**Review by DGamerL — CHANGES_REQUESTED** [review]
Please check if this still works for zippos

File: code/game/objects/items/weapons/lighters.dm (hunk 6 above)
Comment (Spaghetti-bit) [reply code/game/objects/items/weapons/lighters.dm]: Moved to `/obj/item/lighter/random`. Making this still work with zippos while also not cluttering the zippos.

**Review by DGamerL — APPROVED** [review]

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**